tests/st/ipam/test_ipam.py

```diff
@@ -0,0 +1,255 @@
+        cls.hosts.append(DockerHost("host2",
+                                    additional_docker_options=ADDITIONAL_DOCKER_OPTIONS,
+                                    post_docker_commands=POST_DOCKER_COMMANDS,
+                                    start_calico=False))
+        cls.hosts[0].start_calico_node()
+        cls.hosts[1].start_calico_node()
+        cls.network = cls.create_network(cls.hosts[0], "testnet1")
+        cls.workloads = []
+
+    @classmethod
+    def tearDownClass(cls):
+        # Tidy up
+        for host in cls.hosts:
+            host.remove_workloads()
+        cls.network.delete()
+        for host in cls.hosts:
+            host.cleanup()
+            del host
+
+    def setUp(self):
+        # Save off original pool if any, then wipe pools so we have a known ground state
+        response = self.hosts[0].calicoctl("get IPpool -o yaml")
+        self.orig_pools = yaml.safe_load(response)
+        if len(self.orig_pools) > 0:
+            self.hosts[0].writefile("orig_pools.yaml", response)
+            self.hosts[0].calicoctl("delete -f orig_pools.yaml")
+
+    def tearDown(self):
+        # Replace original pool, if any
+        if len(self.orig_pools) > 0:
+            self.hosts[0].calicoctl("apply -f orig_pools.yaml")
+
+    def test_pools_add(self):
+        """
+        (Add a pool), create containers, check IPs assigned from pool.
+        Then Delete that pool.
+        Add a new pool, create containers, check IPs assigned from NEW pool
+        """
+        ipv4_subnet = netaddr.IPNetwork("192.168.0.0/24")
+        new_pool = {'apiVersion': 'v1',
+                    'kind': 'ipPool',
+                    'metadata': {'cidr': str(ipv4_subnet.ipv4())},
+                    }
+        self.hosts[0].writefile("newpool.yaml", yaml.dump(new_pool))
+        self.hosts[0].calicoctl("create -f newpool.yaml")
+
+        for host in self.hosts:
+            workload = host.create_workload("wlda-%s" % host.name,
+                                            image="workload",
+                                            network=self.network)
+            assert netaddr.IPAddress(workload.ip) in ipv4_subnet
+
+        self.hosts[0].calicoctl("delete -f newpool.yaml")
+
+        ipv4_subnet = netaddr.IPNetwork("10.0.1.0/24")
+        new_pool = {'apiVersion': 'v1',
+                    'kind': 'ipPool',
+                    'metadata': {'cidr': str(ipv4_subnet.ipv4())},
+                    }
+        self.hosts[0].writefile("pools.yaml", yaml.dump(new_pool))
+        self.hosts[0].calicoctl("create -f pools.yaml")
+
+        for host in self.hosts:
+            workload = host.create_workload("wlda2-%s" % host.name,
+                                            image="workload",
+                                            network=self.network)
+            assert netaddr.IPAddress(workload.ip) in ipv4_subnet, \
+                "Workload IP in wrong pool. IP: %s, Pool: %s" % (workload.ip, ipv4_subnet.ipv4())
```

Comment:
As discussed, would be great to have a check that, after removing a pool and all pods in the pool, no stale routes are present.